twine/cli.py

```diff
@@ -70,4 +69,4 @@ def dispatch(argv: List[str]) -> Any:
 
     main = registered_commands[args.command].load()
 
-    return main(args.args)
+    return bool(main(args.args))
```

Comment:
This is a change in behavior, but I think it results in the same outcome in `__main__`, where it's used as the argument to `sys.exit`. It felt cleaner than returning an `Optional[bool]`, which I did in https://github.com/pypa/twine/pull/613/commits/26357ba7e1955c704776d0c9406410947a91e78a.

Reply:
What if we just never returned anything? What's returning a bool? Why would we want to return a bool here?

Reply:
To be clear, I'd like to keep `return` here to signal the end of execution but ensure that any sub-command we have doesn't return anything

Reply:
Ah, I could have been clearer there. I added a comment, but `register()` and `upload()` returns `None`, while `check()` has returned `bool` since it was added in #395.

https://github.com/pypa/twine/blob/da8d62f6e8878900267a8499d4e3cfe49ac11002/twine/commands/check.py#L107

Should that change?

Reply:
I'd rather bring that in-line with everything else. I'm open to being convinced otherwise, but I see no value in returning a bool here

Reply:
Are you suggesting that `check()` should return `None`? I think that would break release pipelines that are expecting a non-zero exit code if `twine check` fails. Since its inception, the `bool` has indicated "failure", and ends up as the argument to `sys.exit()`.

https://github.com/pypa/twine/blob/da8d62f6e8878900267a8499d4e3cfe49ac11002/twine/__main__.py#L23-L31

Just to check, I read up on [sys.exit](https://docs.python.org/3/library/sys.html#sys.exit), and verified the shell behavior:

```
$ for result in None bool(None) False True; do
> echo -n "$result: "
> python3 -c "import sys; sys.exit($result)"
> echo $?
> done
None: 0
bool(None): 0
False: 0
True: 1
```

That said, I'm not married to the `bool` cast, and depending on where one is reading, it might not be obvious that `True` means "failure". An alternative is what I did in https://github.com/pypa/twine/pull/613/commits/26357ba7e1955c704776d0c9406410947a91e78a, which is to basically return an `Optional[bool]`. Or, I could take the original approach, and return `Any`. Or ???

Reply:
Why not return the actual thing instead of a bool? Yes `sys.exit(bool)` works but only because True and False have bad default behaviour on most python's of behaving int-ish in certain scenarios. We should be returning `0`, `1`, or something else and _explicitly_ documenting those error codes

Reply:
After thinking about this a more, and reading your comment, plus this in  `sys.exit` docs:

> The optional argument `arg` can be an integer giving the exit status (defaulting to zero), or another type of object.... If another type of object is passed, `None` is equivalent to passing zero, and any other object is printed to `stderr` and results in an exit code of 1.

I've reverted back to returning `Any` in https://github.com/pypa/twine/pull/613/commits/3034635359417b04c820c1ac5e197542e7772734. This feels like the most accurate representation of the current behavior, which I've generally tried to preserve as much as possible in all of the typing work.

If we want to change the return value of `check.main` and/or `cli.dispatch` to `int`, and/or define explicit error codes, I'd rather address that in a separate issue.